In [2]:
import numpy as np
from scipy import stats
import pandas as pd
import statsmodels.api as sm
import matplotlib.pyplot as plt
import seaborn as sns
from statsmodels.stats.proportion import proportions_ztest
from scipy.stats import shapiro

### p45
- 자동차의 1리터 당 주행거리 평균 = 12, 표준편차 = 3인 정규분포를 따른다.
- P(9≤X≤15) = ? 
- P(15≤X) = ? 

In [2]:
mu = 12
sigma = 3
x1 = 15
prob1 = stats.norm.cdf(x1, mu, sigma)  # 누적확률이 15이하 인값 = prob1

# 9km 이상 15km 이하를 달릴 확률은 (-1시그마에서 1시그마범위) 0.341+0.341 = 0.683 이다.
print("{0}km 이상 {1}km 이하를 달릴 확률은 {2:.3f}%에 해당함".format(mu-3,x1,(prob1-0.5)*2))

# 15km 이상을 달릴 확률은 (1 - 1시그마까지 누적확률) 1-0.841 = 0.159 이다.
print("{0}km 이상을 달릴 확률은 {1:.3f}%에 해당함".format(x1,(1-prob1)))

9km 이상 15km 이하를 달릴 확률은 0.683%에 해당함
15km 이상을 달릴 확률은 0.159%에 해당함


#### 해석
- 해당 확률변수가 N(12, $3^2$)인 정규분포를 따를 때, P(9≤X≤15) = 0.68는 1리터 당 주행거리를 9km에서 15km를 달릴 확률은 0.68로 68%이다.

- 해당 확률변수가 N(12, $3^2$)인 정규분포를 따를 때, P(15≤X) = 0.16는 1리터 당 주행거리를 15km 이상을 달릴 확률은 0.16로 약 16%이다.


### p78

In [5]:
mu = 3

# 1시간에 2명의 손님이 도착할 확률
prob = stats.poisson.pmf(2, mu)

# 1시간에 1명이하의 손님이 도착할 확률   (1시간에 0명일때 + 1시간에 1명일때) 의 개념이므로 pmf말고 cdf 사용함
cdf_prob = stats.poisson.cdf(1, mu)  

# 1시간에 3명이상의 손님이 도착할 확률   ( 1 - 2명이하 손님이 확률)
cdf_prob2 = stats.poisson.cdf(2, mu)

print('10시에서 11시사이에서 {0}명의 손님이 도착할 확률 : {1:.2f}%'.format(2, prob))
print('10시에서 11시사이에서 {0}명 이하의 손님이 도착할 확률 : {1:.2f}%'.format(1,cdf_prob))
print('10시에서 11시사이에서 {0}명 이상의 손님이 도착할 확률 : {1:.2f}%'.format(3,1-cdf_prob2))

0.42319008112684364
10시에서 11시사이에서 2명의 손님이 도착할 확률 : 0.22%
10시에서 11시사이에서 1명 이하의 손님이 도착할 확률 : 0.20%
10시에서 11시사이에서 3명 이상의 손님이 도착할 확률 : 0.58%


### p96

In [7]:
from scipy.stats import shapiro

ds = pd.read_csv('C:/Users/USER/Desktop/포스코 청년 아카데미_B4_김민경/통계이해/통계이해자료/2sample_t_test.csv')
ds.head()

ds = ds.dropna()  # 결측치를 무시하라
data1 = ds['Mat_A']
data2 = ds['Mat_B']


# 정규성 검정
shapiro(data1)
shapiro(data2)

statistic, p = stats.shapiro(data1)
statistic2, p2 = stats.shapiro(data2)

print("Shapiro-Wilk Test1 : statistic = {}, p-value = {}".format(statistic, p))

print("Shapiro-Wilk Test2 : statistic = {}, p-value = {}".format(statistic2, p2)) 

# Shapiro-Wilk Test1 은 검정 통계랑이  0.906, p-value 값이 0.256 이므로 p-value > 0.05 
# 따라서 귀무가설을 채택한다. 
# 표본으로 수집된 data는 정규분포를 따른다고 볼 수 있다. 

# Shapiro-Wilk Test2 은 검정 통계랑이  0.972, p-value 값이 0.916 이므로 p-value > 0.05 
# 따라서 귀무가설을 채택한다. 
# 표본으로 수집된 data는 정규분포를 따른다고 볼 수 있다. 

Shapiro-Wilk Test1 : statistic = 0.9062638878822327, p-value = 0.25630074739456177
Shapiro-Wilk Test2 : statistic = 0.9729756116867065, p-value = 0.9169799089431763


#### Shapiro-Wilk Test1 해석
H0: 모집단이 정규분포를 따른다. <br>
H1: 모집단이 정규분포를 따르지 않는다.<br>
라는 가설을 설정하고 소규모 집단의 샤피로-윌크 검정을 하였을 때, 누락값을 제외한 Mat_A에 대한 검정통계량은 0.91이며 p-value는 0.256이 나왔다.<br>
유의수준 $\alpha$ = 0.05(95% 신뢰수준)에서 검정 결과, p-value > $\alpha$로 귀무가설 H0를 기각하지 못한다.<br>
따라서, 모집단이 정규분포를 따른다고 할 수 있다.


#### Shapiro-Wilk Test2 해석
H0: 모집단이 정규분포를 따른다. <br>
H1: 모집단이 정규분포를 따르지 않는다.<br>
라는 가설을 설정하고 소규모 집단의 샤피로-윌크 검정을 하였을 때, 누락값을 제외한 Mat_B에 대한 검정통계량은 0.97이며 p-value는 0.92이 나왔다.<br>
유의수준 $\alpha$ = 0.05(95% 신뢰수준)에서 검정 결과, p-value > $\alpha$로 귀무가설 H0를 기각하지 못한다.<br>
따라서, 모집단이 정규분포를 따른다고 할 수 있다.

### p99

# 손으로 계산
평균 x = 19.5    모표준편차 = 3.8
19.5 - 1.96* (3.8/루트 20) <= M <= 19.5 + 1.96* (3.8/루트 20)
17.84 <= M <= 21.165 나왔습니다

In [4]:
import pandas as pd
df = pd.DataFrame({'sample': [18, 18, 20, 21, 20, 23, 19, 18, 17, 21, 
                             22, 20, 20, 21, 20, 19, 19, 18, 17, 19]})
df.head()

# p-value > alpha로 정규성을 따른다.
print(shapiro(df))

# 정규분포 구간 추정 함수 실행 
lower, upper = stats.norm.interval(0.95, loc = np.mean(df), scale = 3.8/np.sqrt(20))
print("신뢰구간 : ({0},{1})".format(lower.round(2),upper.round(2)))

# 모평균 = 19.5 / 모평균이 신뢰구간인 7.83 에서 21.17 안에 포함된다. 즉 17.8~21.17 이 구간안에서 모평균을 추정할 수 있다.

ShapiroResult(statistic=0.955651044845581, pvalue=0.46101993322372437)
신뢰구간 : ([17.83],[21.17])


#### 해석
먼저, H0: 모집단은 정규성을 따른다. H1: 모집단은 정규성을 따르지 않는다.는 가설 설정 후 정규성에 대해 검정한 결과,<br>
p-value = 0.46 > alpha = 0.05로 귀무가설을 채택한다. 따라서 정규성을 따른다고 할 수 있다. <br>
모표준편차를 알고있을 때, 모평균의 신뢰구간을 추정하려면 z-test를 이용해야한다. <br>
새로운 품종의 감자에 대한 평균 무게의 95% 신뢰구간을 구하면 17.83 ≤ $\mu$ ≤ 21.17이다. <br>
(단위 정보 없음 - kg으로 판단) 17.83kg와 21.17kg 구간 내에 새 품종 감자의 모평균이 존재한다.

### p102

In [6]:
import pandas as pd
from scipy.stats import shapiro
df = pd.DataFrame({'sample': [73, 71, 74, 69, 70,73, 70, 68, 75, 72,
                             70, 72, 73, 70, 70, 72, 71, 70, 75, 72]})
df.head()

print(shapiro(df))


# 정규분포 구간 추정 함수 실행
lower, upper = stats.t.interval(0.95, len(df)-1,loc = np.mean(df), scale = stats.sem(df))
print("신뢰구간 : ({0},{1})".format(lower.round(2),upper.round(2)))

# 평균 인장강도를 70kg/mm2 으로 기대하고 있다. 
# 신뢰구간이 70.6 ~ 72.4 이므로 기대한 평균이 신뢰구간에 들어와 있기 때문에 새로 개발된 강종의 평균 인장을 70으로 볼 수 있다.

ShapiroResult(statistic=0.9470751285552979, pvalue=0.3248629868030548)
신뢰구간 : ([70.6],[72.4])


#### 해석
먼저, H0: 모집단은 정규성을 따른다. H1: 모집단은 정규성을 따르지 않는다.는 가설 설정 후 정규성에 대해 검정한 결과,<br>
p-value = 0.32 > alpha = 0.05로 귀무가설을 채택한다. 따라서 정규성을 따른다고 할 수 있다. <br>
모표준편차를 모를 때, 모평균의 신뢰구간을 추정하려면 t-test를 이용해야한다. <br>
새로운 강종의 평균 인장강도에 대해 95% 신뢰구간을 구하면 70.6 ≤ $\mu$ ≤ 72.4이다. <br>
즉, 70.6kg/$mm^2$와 72.4kg/$mm^2$ 구간 내에 새로운 강종의 인장강도에 대한 모평균이 존재한다.